src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -203,9 +203,17 @@ private BoundExpression ConvertCaseExpression(TypeSymbol switchGoverningType, CS
                 convertedCaseExpression = this.CreateConversion(caseExpression, conversion, switchGoverningType, diagnostics);
             }
 
-            if (switchGoverningType.IsNullableType() && convertedCaseExpression.Kind == BoundKind.Conversion)
+            if (switchGoverningType.IsNullableType()
+                && convertedCaseExpression.Kind == BoundKind.Conversion
+                // Null is a special case here because we want to compare null to the Nullable<T> itself, not to the underlying type.
```

Comment:
Is there an existing unit-test that was failing without the special case for the null value? If not, please add one.